### DISCUSSION TITLE
Passwords hashed?

### DISCUSSION THREAD

**Discussion by dHanel** [serverpod/serverpod#1393] [Q&A] [answered]
Does Serverpod store passwords for user authentication hashed and salted or in plaintext?

**Comment by IzenDev** [comment, marked as answer]
@dHanel Serverpod stores password's hashed and salted.
Below is the method to `generatePasswordHash()`.
![Screenshot 2023-09-24 at 5 32 44 PM](https://github.com/serverpod/serverpod/assets/32414659/baeba331-2ac9-4212-a51b-a6aeaad07674)
